plugins/UFPWriter/UFPWriter.py

```diff
@@ -60,5 +60,43 @@ def write(self, stream, nodes, mode = MeshWriter.OutputMode.BinaryMode):
         else:
             Logger.log("d", "Thumbnail not created, cannot save it")
 
+        #Store the material.
+        application = Application.getInstance()
+        machine_manager = application.getMachineManager()
+        global_stack = machine_manager.activeMachine
+        container_registry = application.getContainerRegistry()
+
+        archive.addContentType(extension="xml.fdm_material", mime_type="application/x-ultimaker-material-profile")
+
+        added_materials = []
+        for extruder_stack in global_stack.extruders.values():
+            material = extruder_stack.material
+            material_file_name = material.getMetaData()["base_file"] + ".xml.fdm_material"
+            material_file_name = "/Cura/" + material_file_name
+
+            #Same material cannot be added
+            if material_file_name in added_materials:
+                continue
+
+            material_file = archive.getStream(material_file_name)
```

Comment:
There's not point in already opening the stream if in the next step the container cannot be found.